### PR TITLE
Add specs for IPAM search closest match prefix

### PR DESCRIPTION
**Pull request by gmazoyer**
# Why

Adds specifications for Spec Kit for IPAM search closest matched prefix for non-existing IP INFP-431

## What changed

* Added markdown files for IPAM specification

## How to review

- I opted for no cap to return all containing prefixes, ordered most specific first, I would imagine there would not be so many nesting levels that we could not handle them properly.
- I opted to only show prefix lookup when an IP or prefix is detected but I suppose it's a matter of taste.

## Summary by CodeRabbit

* **Documentation**
  * Added a Specification Quality Checklist confirming readiness and coverage for the IPAM Parent Prefix Lookup feature; all checklist items marked complete.
  * Added a comprehensive feature specification for IPAM Parent Prefix Lookup covering user stories, acceptance criteria, functional requirements, UI behavior (dedicated "Parent Prefixes" results and navigation), edge cases, visibility/normalization rules, and testing guidance.

### CODE REVIEW

File: dev/specs/infp-431-ipam-closest-prefix/spec.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+**Independent Test**: Can be fully tested by searching for any valid IP address in the search dialog and verifying that containing prefixes are returned. Delivers immediate value by eliminating manual prefix calculation.
+
+**Acceptance Scenarios**:
+
+1. **Given** prefixes 10.0.0.0/8, 10.1.0.0/16, and 10.1.2.0/24 exist in the system, **When** the user searches for "10.1.2.45" in search anywhere, **Then** all three containing prefixes are returned, ordered from most specific (10.1.2.0/24) to least specific (10.0.0.0/8), each showing its namespace.
+2. **Given** no prefix contains the searched IP address, **When** the user searches for "192.168.1.1" and no matching prefix exists, **Then** the system returns no results (empty state) with an appropriate message.
+3. **Given** the same IP address exists within prefixes in multiple namespaces, **When** the user searches for that IP address, **Then** results from all namespaces are returned, each clearly labeled with its namespace.
```
Comment (wvandeun) [inline dev/specs/infp-431-ipam-closest-prefix/spec.md]: Can we clarify here what the behavior is going to be when the user searches for a valid IP address that exists in the system? Today we just return the matching IP address objects.

We should probably also add that other existing search anywhere behavior should not be altered by this implementation?

File: dev/specs/infp-431-ipam-closest-prefix/spec.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+- **FR-002**: System MUST return all prefixes that contain the searched IP address or prefix, across all IP namespaces, with no cap on result count.
+- **FR-003**: System MUST order results by prefix specificity, with the most specific (longest) prefix first.
+- **FR-004**: System MUST display the namespace for each returned prefix result so users can distinguish between overlapping IP spaces.
+- **FR-005**: System MUST fall back to existing text search behavior when the query is not a valid complete IP address or CIDR prefix (e.g., partial IPs, hostnames, or general text).
+- **FR-006**: System MUST normalize IPv6 input before performing the lookup to ensure consistent matching regardless of input formatting variations.
+- **FR-007**: System MUST support both IPv4 and IPv6 addresses and prefixes.
+- **FR-008**: System MUST display parent prefix lookup results in a dedicated "Parent Prefixes" section within the search anywhere UI, visually separated from regular text search results, so users can clearly identify containment-based results.
```
Comment (wvandeun) [inline dev/specs/infp-431-ipam-closest-prefix/spec.md]: @bilalabbad @pa-lem this does imply frontend work

**Review by wvandeun — APPROVED** [review]
